src/ubuntu-docker/lind-base/Dockerfile

```diff
@@ -0,0 +1,131 @@
+FROM amd64/ubuntu:20.04
+LABEL lind "v1.0-lind-base-ubuntu-18.04"
+LABEL description "Lind Toolchain (Base)"
+LABEL author "Nicholas Renner nrenner@nyu.edu"
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+ADD etc/ /etc/
+ADD home/ /home/
+ADD root/ /root/
+
+USER root
+
+WORKDIR /
+RUN useradd -Um -s/bin/bash -u1000 lind
+RUN mkdir -pv "/root/.local/etc/" "/home/lind/.local/etc/"
+RUN chown -LR lind:lind "/home/lind/"
+RUN chown -LR root:root "/etc/" "/root/"
+
+ADD make-3.82/ /home/make-3.82
+
+RUN apt update
+RUN apt install build-essential -y
+RUN apt install -y software-properties-common wget
+RUN apt install -y gcc-7 g++
+# RUN wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
+# RUN add-apt-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal main'
+# RUN echo "deb-src http://apt.llvm.org/focal/ llvm-toolchain-focal main" >> /etc/apt/sources.list
+RUN apt-get update -y
+# RUN wget https://apt.llvm.org/llvm.sh
+# RUN chmod +x llvm.sh
+# RUN ./llvm.sh 19
+# RUN apt-get install clang lldb -y
+# RUN apt-get install clang-format clang-tidy clang-tools clang clangd libc++-dev libc++1 libc++abi-dev libc++abi1 libclang-dev libclang1 libllvm-ocaml-dev libomp-dev libomp5 lld lldb llvm-dev llvm-runtime llvm -y
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    git mercurial rsync perl lynx ltrace \
+	w3m gawk procps ruby strace gdb reptyr ed \
+  	vim elfutils libelf-dev gdb less \
+  	netcat htop bc ctags \
+ 	libncurses-dev gcc-multilib gperf diffutils \
+	google-perftools texinfo fortune-mod gnupg highlight fasd \
+	python2.7 python2.7-dev p7zip fuse ack \
+	isc-dhcp-client iputils-ping iproute2 \
+ 	net-tools openresolv openssh-server subversion \
```

Comment:
Just curious, how do we know these are the libraries that we need for this Docker container?

Reply:
we have referenced them from the arch linux image and what libraries it was using. although we could do a analysis and figure out some redundancies to reduce image size.

Reply:
Got it.